cmd/kubeadm/app/componentconfigs/registrations.go

```diff
@@ -65,6 +71,14 @@ func (r Registration) Unmarshal(fileContent []byte) (runtime.Object, error) {
 	// Do a deepcopy of the empty value so we don't mutate it, which could lead to strange errors
 	obj := r.EmptyValue.DeepCopyObject()
 
+	// Check for duplicate and unknown fields using yaml.UnmarshalStrict().
+	// Use the external config as a target, because external configs can have JSON tags that do not match
+	// the field names - e.g. KubeletConfig's 'resolvConf'
+	objExternal := r.EmptyValueExternal.DeepCopyObject()
```

Comment:
this means only a single external version is supported, right? Doesn't this lose the benefit of supporting multiple versions allowing evolution of the serialized config?

Reply:
i guess that's true. i can think of a workaround for that using a map.

do you have context of why was the resolvConf "tag vs field name" difference was allowed in the KubeletConfig? this prevents us from unmarshaling into the internal config.

Reply:
internal objects must *never* be used as deserialization targets. even if field names currently match, they are not guaranteed to be stable. That is the point of the external types and json tags.

Reply:
ok, understood.